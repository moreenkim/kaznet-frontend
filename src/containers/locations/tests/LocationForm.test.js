// Test LocationForm
import React from "react";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";

import ErrorBoundary from "../../../components/ErrorBoundary";
import { LocationForm } from "../LocationForm";
import * as fixtures from "../../../store/locations/tests/fixtures";
import { locationTypesById } from "../../../store/locationTypes/tests/fixtures";

describe("containers/location/LocationForm", () => {
  it("renders without crashing", () => {
    const initialData = {
      name: "Kitui",
      parent: "",
      location_type: "",
      geopoint: "36,-1",
      radius: "9",
      shapefile: ""
    };
    shallow(
      <LocationForm
        formActionDispatch={function() {}}
        fetchLocations={function() {}}
        fetchLocationTypes={function() {}}
        initialData={initialData}
        locationsById={fixtures.locationsById}
        locationTypesById={locationTypesById}
      />
    );
  });

  it("renders location form correctly", () => {
    const initialData = {
      name: "Kitui",
      parent: "",
      location_type: "",
      geopoint: "36,-1",
      radius: "9",
      shapefile: ""
    };
    const wrapper = mount(
      <ErrorBoundary>
        <LocationForm
          formActionDispatch={function() {}}
          fetchLocations={function() {}}
          fetchLocationTypes={function() {}}
          initialData={initialData}
          locationsById={fixtures.locationsById}
          locationTypesById={locationTypesById}
        />
      </ErrorBoundary>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.unmount();
  });
});
