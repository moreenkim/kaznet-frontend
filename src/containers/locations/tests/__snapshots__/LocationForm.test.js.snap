// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/location/LocationForm renders location form correctly 1`] = `
<ErrorBoundary>
  <h1>
    Something went wrong.
  </h1>
</ErrorBoundary>
`;
